modules/chapter1/pages/section2.adoc

```diff
@@ -529,4 +544,155 @@ We're not going to go through the code and configuration for this, but here is t
+An example of this is in the https://github.com/kubeflow/kfp-tekton/blob/master/sdk/python/tests/compiler/testdata/artifact_passing_using_volume.py[KFP compiler tests] which we will discuss here.
+
+First create the volume to be used and assign it to a variable:
+[source,python]
+include::example$artifact_passing_using_volume.py[lines=78..79]
+
+[source,python]
+include::example$artifact_passing_using_volume.py[lines=81..88]
+
+Then add definition to the _pipeline configuration_:
+[source,python]
+include::example$artifact_passing_using_volume.py[lines=91..93]
+
+
+[IMPORTANT]
+====
+The *_data-volume PVC claim_* needs to exist on the OpenShift cluster before running the pipeline, if not then the _pipeline execution pod_ will fail to deploy and the run will terminate.
+====
+
+To pass big data using cloud provider volumes, it's recommended to use the *_volume_based_data_passing_method_*.
+
+== Data Science Pipeline - Hands On Example
+
+We're now going to deploy and run an example DSP pipeline. The use case is offline scoring for _Fraud Detection_.
+
+=== Setup
+
+[%interactive]
+* [ ] Using the Minio UI, create a new S3 Bucket named _fraud_detection_bucket_.
+
+image::create-s3-fraud-bucket.png[]
+
+[%interactive]
+* [ ] Download the following artifacts:
+
+** [ ] xref:attachment$offline_scoring_kfp_pipeline.py[ONNX Model]
+** [ ] xref:attachment$live-data.csv.gz[Live Transaction Data]
+** [ ] xref:attachment$training-data.csv.gz[Model Training Data]
+
+[%interactive]
+* [ ] Uncompress the *_.gz_* files using the _gzip -d _ command. 
+* [ ] Upload the files to S3 _fraud_detection_bucket_ via the _Minio_ UI.
+
+====
+image::fraud-upload-files.png[]
+====
+====
+image::fraud-files-uploaded.png[]
+====
+
+[%interactive]
+* [ ] Create the *_aws-connection-fraud-detection_* _secret_ containing the _S3_ bucket name, endpoint and credentials.
+
+[source,bash]
+--
+oc create secret generic aws-connection-fraud-detection --from-literal=AWS_ACCESS_KEY_ID=minio --from-literal=AWS_DEFAULT_REGION=rhods-enablement-emea --from-literal=AWS_S3_BUCKET=fraud-detection-bucket --from-literal=AWS_S3_ENDPOINT=https://minio-api... --from-literal=AWS_SECRET_ACCESS_KEY=minio123
+--
+
+[%interactive]
+* [ ] Create two OpenShift Persistent Volumes with the following names:
+** *offline-scoring-model-volume*
+** *offline-scoring-data-volume*
+
+TIP: Example PVC definitions are here, download and customise as necessary this xref:attachment$offline-scoring-data-pvc.yaml[Data Volume definition] & xref:attachment$offline-scoring-model-pvc.yaml[Model Volume definition]
+
+
+=== Execute Pipeline
+
+Download and customise as necessary this xref:attachment$offline_scoring_kfp_pipeline.py[Example Volume definition]
+
+[%interactive]
+* [ ] Create a _Workbench_ with the name _fraud-onnx_
+
+image::fraud-onnx-workbench.png[]
+
+[%interactive]
+* [ ] Download the xref:attachment$offline_scoring_kfp_pipeline.py[Offline Scoring Pipeline] and load it into your _workbench_.
```

Comment:
I recommend using the course's git repository and asking the student to just check it out and access the code.

Reply:
The screenshot shows the code inside of a Jupyter notebook. The code is written to be executed through the CLI (see `+if __name__=='__main__'+` clause). Of course, it can be executed within a notebook, but then let's provide it in the notebook format.

Reply:
I'd rather keep the code in a single repo rather than splitting it across two repos